Automattic/wp-calypso
# wp-admin Site Default: prevent race condition on admin interface card

- Closes https://github.com/Automattic/dotcom-forge/issues/4291

## Proposed Changes

- Freeze the options when the mutation is updating to avoid the user changing it multiple times at once.
- Update redux store instead of requesting the whole site info on success.
- Some react query optimizations.

## Testing Instructions

1. Navigate to `/hosting-config/$siteSlug` on the current branch.
2. Scroll down to locate the `Admin Interface Style` section.
3. Select `Classic Style` from the available options.
4. Note that the options will be locked, preventing any changes until the mutation process is complete.
5. Try changing the option multiple times.
6. Confirm that the option consistently displays a single value without fluctuating, thereby ensuring that no race condition exists.


## Screencast


https://github.com/Automattic/wp-calypso/assets/779993/f784d149-1549-4919-bbc9-64f6e905e549



## Pre-merge Checklist
